contacts/main.js

```diff
@@ -0,0 +1,566 @@
+    if (p1FullName < p2FullName) return -1;
+    return 0;
+  })
+}
+
+function rerenderFavsListHTML(mapOfContacts) {
+
+  let ol = document.getElementById('favs-list');
+  ol.innerHTML = '';
+
+  let numbers = Object.keys(mapOfContacts).sort();
+  numbers.reverse();
+
+  for (let i = 0; i < numbers.length; i++) {
+    if (i === 3) break;
+
+    let p = document.createElement('p');
+    p.innerHTML = numbers[i];
+    p.className = 'align-center';
+    ol.appendChild(p);
+
+    mapOfContacts[numbers[i]].forEach((item) => {
+
+      let newLi = document.createElement('li');
+      newLi.classList.add('person-field');
+
+      let fav_count = document.createElement('div');
+      fav_count.className = 'favs-count'
+      fav_count.innerHTML = item.likes;
+      newLi.appendChild(fav_count);
+
+      let image = document.createElement('img');
+      image.src = item.image;
+      image.alt = 'pic';
+      image.className = 'avatar';
+      newLi.appendChild(image);
+
+      let full_name = document.createElement('span');
+      full_name.className = 'full-name';
+      let first_name = document.createElement('span');
+      first_name.innerHTML = item.firstName;
+      first_name.style.marginRight = '10px';
+      full_name.appendChild(first_name);
+      let last_name = document.createElement('span');
+      last_name.innerHTML = item.lastName;
+      full_name.appendChild(last_name);
+      newLi.appendChild(full_name);
+
+      ol.appendChild(newLi);
+    })
+  }
+}
+//-----------------------------actions----------------------------------
+function deleteAllButtonAction() {
+  if (confirm('ARE YOU SURE YOU WANT TO DO THIS ?')) {
+    deleteSelectedContacts();
+    refreshFavsList();
+    rerenderContacts();
+    resetDeleteAllButtonValueAndState();
+  } else {
+    clearSelectedBoxes();
+    resetDeleteAllButtonValueAndState();
+  }
+}
+//----------------------------------------------------------------------
+
+function deleteSelectedContacts() {
+  let personsToDelete = getSelectedContactsIDs();
+  for (let i = 0; i < personsToDelete.length; i++) {
+    deletePerson(personsToDelete[i]);
+  }
+}
+
+function getSelectedContactsIDs() {
+  // returns an array of ids
+  let returnArrayOfIDs = [];
+  let arr = document.getElementById('contacts-list').
+  getElementsByClassName('person-field');
+  for (let i = 0; i < arr.length; i++) {
+    let input = arr[i].getElementsByTagName('input')[0];
+    if (input.type === 'checkbox' && input.checked) {
+      // compute the id -> last 6 characters
+      let personId = input.id.slice(input.id.length - 6);
+      returnArrayOfIDs.push(personId);
+    }
+  }
+  return returnArrayOfIDs;
+}
+
+function clearSelectedBoxes() {
+  let ch = getSelectedContactsIDs();
+  ch.forEach((id) => {
+    let buttonId = 'checkbox-' + id;
+    document.getElementById(buttonId).checked = false;
+  })
+}
+
+///////////////////////validation functions///////////////////////
+function validateFirstNameInput(firstName) {
```

Comment:
use a single method, validateName, since validateFirstNameInput and validaetLastNameInput do the same thing

Reply:
i know, but symmetry looks good :)))

Reply:
looks good... but there is more 'duplicate' code to maintain